studio/components/interfaces/Settings/Logs/LogsPreviewer.tsx

```diff
@@ -83,24 +83,48 @@ export const LogsPreviewer: React.FC<Props> = ({
       },
     })
   }
-
-  const handleSearch: LogSearchCallback = async ({ query = '', to, from }) => {
-    setParams((prev) => ({
-      ...prev,
-      iso_timestamp_start: from || prev.iso_timestamp_start || '',
-      iso_timestamp_end: to || prev.iso_timestamp_end || '',
-    }))
-    setFilters((prev) => ({ ...prev, search_query: query }))
-    router.push({
-      pathname: router.pathname,
-      query: {
-        ...router.query,
-        s: query || '',
-        its: from || its || '',
-        ite: to || ite || '',
-      },
-    })
+  const handleSearch: LogSearchCallback = async (event, { query, to, from }) => {
+    if (event === 'search-input-change') {
+      setFilters((prev) => ({ ...prev, search_query: query }))
+      router.push({
+        pathname: router.pathname,
+        query: { ...router.query, s: query },
+      })
+    } else if (event === 'event-chart-bar-click') {
+      const [nextStart, nextEnd] = ensureNoTimestampConflict(
+        [params.iso_timestamp_start || '', params.iso_timestamp_end || ''],
+        [from || '', to || '']
+      )
+      setParams((prev) => ({
+        ...prev,
+        iso_timestamp_start: nextStart,
+        iso_timestamp_end: nextEnd,
+      }))
+      router.push({
```

Comment:
btw if it might help simplify for future refactor - we now have a useUrlState hook which should be able to help with this